static/js/actions/email.js

```diff
@@ -33,13 +39,30 @@ export function sendSearchResultMail(
   searchRequest: Object
 ): Dispatcher<EmailSendResponse> {
   return (dispatch: Dispatch) => {
-    dispatch(initiateSendEmail());
+    dispatch(initiateSendEmail(SEARCH_EMAIL_TYPE));
     return api.sendSearchResultMail(subject, body, searchRequest).
       then(response => {
-        dispatch(sendEmailSuccess());
+        dispatch(sendEmailSuccess(SEARCH_EMAIL_TYPE));
         return Promise.resolve(response);
       }).catch(error => {
-        dispatch(sendEmailFailure(error));
+        dispatch(sendEmailFailure({type: SEARCH_EMAIL_TYPE, error: error}));
+      });
+  };
+}
+
+export function sendCourseTeamMail(
+  subject: string,
+  body: string,
+  courseId: number
+): Dispatcher<EmailSendResponse> {
+  return (dispatch: Dispatch) => {
+    dispatch(initiateSendEmail(COURSE_EMAIL_TYPE));
+    return api.sendCourseTeamMail(subject, body, courseId).
+      then(response => {
+        dispatch(sendEmailSuccess(COURSE_EMAIL_TYPE));
+        return Promise.resolve(response);
+      }).catch(error => {
+        dispatch(sendEmailFailure({type: COURSE_EMAIL_TYPE, error: error}));
```

Comment:
It seems like it would fruitful (from a code re-use point of view) to refactor the `sendCourseTeamMail` and `sendSearchResultMail` into one function taking a `courseEmailType` - this could even just be dispatched to the API function and handled there. It looks like at the API level we have different request bodies and different URLs to post to, but otherwise we could eliminate a little boilerplate.

Not sure if it makes more sense to work on this here or in #2446